yt_dlp/extractor/rokfin.py

```diff
@@ -0,0 +1,561 @@
+# coding: utf-8
+from json import dumps
+from math import ceil
+import itertools
+import datetime
+from re import finditer
+from urllib.parse import urlparse
+from .common import InfoExtractor, SearchInfoExtractor
+from ..utils import (
+    unified_timestamp,
+    try_get,
+    traverse_obj,
+    int_or_none,
+    bool_or_none,
+    float_or_none,
+    str_or_none,
+    url_or_none,
+    ExtractorError,
+)
+
+
+class RokfinIE(InfoExtractor):
+    _NETRC_MACHINE = 'rokfin'
+
+
+class RokfinSingleVideoIE(RokfinIE):
+    _META_DATA_BASE_URL = 'https://prod-api-v2.production.rokfin.com/api/v2/public/'
+    _RECOMMENDED_VIDEO_BASE_URL = 'https://rokfin.com/'
+    _CHANNEL_BASE_URL = 'https://rokfin.com/'
+
+
+class RokfinPostIE(RokfinSingleVideoIE):
+    IE_NAME = 'rokfin:post'
+    _VALID_URL = r'https?://(?:www\.)?rokfin\.com/(?P<id>post/[^/]+)'
+    _TESTS = [{
+        'url': 'https://www.rokfin.com/post/57548/Mitt-Romneys-Crazy-Solution-To-Climate-Change',
+        'info_dict': {
+            'id': 'post/57548',
+            'ext': 'mp4',
+            'title': 'Mitt Romney\'s Crazy Solution To Climate Change',
+            'thumbnail': 're:https://img.production.rokfin.com/.+',
+            'upload_date': '20211023',
+            'timestamp': 1634998029,
+            'creator': 'Jimmy Dore',
+            'channel': 'Jimmy Dore',
+            'channel_id': 65429,
+            'channel_url': 'https://rokfin.com/TheJimmyDoreShow',
+            'duration': 213.0,
+            'availability': 'public',
+            'live_status': 'not_live'
+        }
+    }]
+
+    def _real_extract(self, url_from_user):
+        if self.get_param('username') or self.get_param('password'):
+            self.report_warning('site logins are unsupported')
```

Comment:
Removing this will cause user-friendliness problem. As you may have seen, someone asked why premium content is not being downloaded, even with valid credentials. That's why I put that in.